### ISSUE TITLE
Legacy Quarg Intrusion Countermeasures should carry a lesser sentence

### ISSUE THREAD

**Issue by ThrawnCA** [endless-sky/endless-sky#8230]
### Problem Description

The upcoming treatment of Quarg outfits as atrocities should perhaps take into consideration the possibility that a player has obtained Intrusion Countermeasures from the Korath (in an earlier game version).

It would still be reasonable to confiscate the outfits and fine the player; receiving stolen property is a crime. But since the player didn't necessarily attack any Quarg to get them, and since the countermeasures are not ridiculously dangerous to galactic peace like Skylances, it probably shouldn't carry a life sentence.

### Related Issue Links

#7737 makes possession of Quarg outfits an atrocity.

### Desired Solution

If the Quarg detect Intrusion Countermeasures, but no other stolen Quarg tech, the player should receive a lesser sentence instead of life imprisonment.

### Alternative Approaches

There could be a location where Quarg outfits can be handed over peacefully, if the player's reputation with the Quarg is positive. If the player has recovered Intrusion Countermeasures from the Korath, handing them back to the Quarg is doing the Quarg a favour, especially compared to the alternative of giving them to an outfitter.

**Comment by ThrawnCA** [comment]
If the player is on a world guarded by Quarg, but without an outfitter, when they update the game, it may be quite difficult for them to ditch their contraband outfits.

**Comment by ThrawnCA** [comment]
Could Intrusion Countermeasures just be dropped entirely from the list of atrocity triggers? No one is likely to bother attacking the Quarg to steal those _and nothing else_. They're nice to have, but they're not game-changers. If someone has no other Quarg gear, chances are that they didn't actually capture a Quarg ship.

**Comment by Hurleveur** [comment]
Their description litteraly says nobody wants to be caught with them.
So if we dont have an atrocity we  should at least make it a serious fine (15 millions?)

The only way to get these in the current version is to capture a Quarg ship which is way more serious than looting it. I dont know how far we should go with supporting old versions, how long ago was that even? Are you sure it was the same name?

**Comment by EjoThims** [comment]
You're confusing legacy data support with in universe lore here.

Your pilot in universe didn't get those from the Korath because the Korath don't have or use them, even though you as the player got them when the gameplay abstraction of that universe included them on Korath vessels as a placeholder before their own counterparts were ready.

Frankly, if this is an issue because your saves are that old, just delete the outfits from your save file instead of asking for game changes to continue supporting your legacy outfits in an unintended way. Consistency of intended form of play is actually best preserved with this simple edit, not by hanging onto outfits you can no longer get without causing the very problems you are now trying to circumvent.

The unintended behavior here is you still having these outfits, not that the outfits you still have aren't grandfathered through for your convenience.

And no, the availability of Pug versions of the same concept has no bearing on how the Quarg react to you having stolen theirs.

**Comment by Hurleveur** [comment]
Actually the mission will NOT trigger if you launch an old save where you have one on a Quarg world. I am sure of that.

**Comment by ThrawnCA** [comment]
> Actually the mission will not trigger if you launch an old save where you have one on a Quarg world. The atrocity check happens when you land, not when you load the save.

I didn't mean a Quarg-owned world, I meant a world that has Quarg ships patrolling the skies, eg the Kor Efreti worlds.

**Comment by Hurleveur** [comment]
The ships dont even scan you???
What are you on about?

**Comment by ThrawnCA** [comment]
> Your pilot in universe didn't get those from the Korath because the Korath don't have or use them, even though you as the player got them when the gameplay abstraction of that universe included them on Korath vessels as a placeholder before their own counterparts were ready.

Nonetheless, a player who updates from an earlier version, where they were more or less legitimately using those outfits, and then suddenly gets jailed for life, even though from their perspective they didn't do anything wrong, may find that jarring.

> The unintended behavior here is you still having these outfits, not that the outfits you still have aren't grandfathered through for your convenience.

Nonetheless, it's likely that anyone who obtained Intrusion Countermeasures in an earlier version _would_ hang onto them, because they are a powerful defence, ensuring that you can disengage if you're ever losing a boarding contest, and they have zero weight (so there's no cost to keeping them) and zero market value (so there's no incentive to dispose of them).

> And no, the availability of Pug versions of the same concept has no bearing on how the Quarg react to you having stolen theirs.

I disagree. The stated motive for tightly restricting their technology is that it's too dangerous to proliferate. If better technology is already available in quantity, that motive goes away.

> ETA: IF we do want to support this equipment in older saves (and I very strongly feel we should -not-, just in case that isn't clear), the best way to do so would actually be to rename and deprecate the Intrusion Countermeasures, not to restrict or modify new development and tool usage in order to perpetually support placeholder usage from outdated game builds.

I don't mind that approach if it's feasible.

> Please do try the behavior to confirm it behaves as you think it does.

I don't have a continuous build. I only learned about this upcoming feature from the wiki.

**Comment by Hurleveur** [comment]
Even if they did (but they have no scanner so they wont), surely if you have managed to acquire a world ship a few probable Quarg ships shouldnt be an issue, they lack range. I truly dont see the issue here honestly.
Even if there was an issue at some poimt, I think that it would be such a niche case that its not worth changing anything.

**Comment by Zireael07** [comment]
Can the "legacy" thing be detected somehow and just have an event delete them from the player ships?

**Comment by Hurleveur** [comment]
@ThrawnCA you should know the conversation was recently updated so that you dont get killed if it happens in flight.
They will just consider it an atrocity.

**Comment by EjoThims** [comment]
> Can the "legacy" thing be detected somehow and just have an event delete them from the player ships?

AFAIK, not without the rename and deprecation that would solve the whole thing anyway.

I'll respond to the longer post again in a bit, but the response really just confirms to me that this is all about holding onto clear placeholders, something that we should not be working in hoops to support.

**Comment by Zireael07** [comment]
Some players started playing long ago (and advanced far enough in the meantime) that "sorry that was a placeholder just edit your savegame" doesn't cut it. So a rename and deprecation should be the way to go imho

**Comment by TomGoodIdea** [comment]
We haven't reached 1.0.0 yet, so we shouldn't complain about lack of backward compatibility.

**Comment by Hurleveur** [comment]
> Some players started playing long ago (and advanced far enough in the meantime) that "sorry that was a placeholder just edit your savegame" doesn't cut it. So a rename and deprecation should be the way to go imho

I think that more testing should be provided to ensure this is indeed a problem. In the meantime I'll mark this unconfirmed.

But I dont see any issues since presumably atrocity/fine check happens when you landed, not as you load.

**Comment by Hurleveur** [comment]
Alr after some investigation:
-no issue in flight because you wont be scanned
-issue is only when you land(not when loading the save)

So I dont see why it should be different from the rest of Quarg tech.
The issue could be we don't receive a warning about it? I feel like it's broader than just the countermeasures. To get them most of that without being at war with the Quarg, you would have had to either cheat, or farm rep. Were the old countermeasures not practically impossible to get, too?

**Comment by warp-core** [comment]
A person loading an old save on a Quarg world with Quarg outfits on their ship won't be immediately affected, and no Quarg ships have scanners at the moment.

The conversation makes it pretty clear what's going on, and I don't think a player should be surprised that things change when they update the game, though perhaps *all* the outfit descriptions shouyld be updated to indicate that they will get you a death sentence if the Quarg catch you with them.

**Comment by Zireael07** [comment]
> The conversation makes it pretty clear what's going on, and I don't think a player should be surprised that things change when they update the game, though perhaps all the outfit descriptions shouyld be updated to indicate that they will get you a death sentence if the Quarg catch you with them.

Oh. Now I get it. I thought just loading an old game would get me in trouble, but since it's only upon landing, yea, let's go for updating descriptions.

**Comment by EjoThims** [comment]
I think updating the descriptions of Quarg outfits to all carry a similar warning makes sense in general, yea.

We should probably also update all their prices to 0. We can add a price back in to the few places that would take the risk when we merge the PR for variable pricing based on location.

**Comment by TomGoodIdea** [comment]
Then we would need to add a new attribute to these outfits or temporarily change prices of Quarg ships, because price is the basis of the combat rating system.

**Comment by EjoThims** [comment]
> Then we would need to add a new attribute to these outfits or temporarily change prices of Quarg ships, because price is the basis of the combat rating system.

It would just mean shifting the total price to the hull itself. The special pricing offered at certain ports (once the PR to do so is merged) wouldn't change the price of ships in flight for those calculations, so there would be no need to switch it back later ever.

**Comment by ThrawnCA** [comment]
> Were the old countermeasures not practically impossible to get, too?

No, it was fairly easy. Korath Raiders all carried them, and the Remnant give you missions to hunt them, so you can kill some of the crew, land and recruit, then come back and finish the job. So it was not hard to farm dozens of Intrusion Countermeasures. I have over a hundred.

**Comment by ThrawnCA** [comment]
On the subject of whether to support older versions, I'll point out that the reason I started on 0.9.8 and upgraded later is because that's the version provided in the Ubuntu repositories. So there will likely be a disproportionate number of people who upgrade from that.

**Comment by Amazinite** [comment]
As Warp described, even being on a Quarg planet when updating the version you're playing on, you won't immediately be game ended by the outfits now being illegal, so we're fine to leave old outfits be.
